packages/main/src/utils/Strings.js

```diff
@@ -0,0 +1,66 @@
+/**
+	* Escape a regular expression text so that it can be used in a regular expression.
+	* @param {string} text The raw text
+	* @returns The regular expression text, with special characters having been escaped.
+	*/
+export function escapeRegex(text) {
+	return text.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+/**
+	* Replace all instances of a particular find text.˝
+	* @param {string} text The text to find/replace on
+	* @param {string} find The string to find
+	* @param {string|(s:string)=>string} replace The replace string or function
+	* @param {boolean=} caseInsensitive true if the replace is case insensitive, false by default
+	* @return {string} The string with all matches replaced.
+	*/
+export function replaceAll(text, find, replace, caseInsensitive) {
+	return text.replace(new RegExp(escapeRegex(find), `${caseInsensitive ? "i" : ""}g`), replace);
+}
+
+// Cache the regular expressions for repeated use
+const markupReplace = `&=&amp;,>=&gt;,<=&lt;,"=&quot;,'=&#39;`.split(",")
+	.map(p => p.split("="))
+	.map(([f, r]) => [new RegExp(escapeRegex(f), "g"), r]);
+
+/**
+	* Escape all html special characters in a string to their html entity equivalent.
+	* @param {string} text The raw text, which may contain markup
+	* @return {string} The escaped text
+	*/
+export function escapeHTML(text) {
+	markupReplace.forEach(([regex, replacement]) => {
+		text = text.replace(regex, replacement);
+	});
+	return text;
+}
+
+/**
+	* Generate markup for a raw string where a particular text is wrapped with some tag, by default `<b>` (bold) tag.
+	* @param {string} text The text to add highlighting to, all characters are interpretted literally (and not considered markup)
+	* @param {string} input The search text to highlight, all characters are interpretted literally (and not considered markup)
+	* @param {string=} highlightStart The start tag, default is "<b>"
+	* @param {string=} highlightEnd The end tag, default is "</b>"
+	* @return {string} the markup HTML which contains all occurrances of the input text surrounded with the markup start/end tags.
+	*/
+export function generateHighlightedMarkup(text, input, highlightStart = "<b>", highlightEnd = "</b>") {
+	if (!text) {
```

Comment:
Probably a check against the type would be helpful, too

Reply:
You mean to say that the input `text` might be something other than a `string` type?